models/src/main/protobuf/RhoTypes.proto

```diff
@@ -42,9 +42,16 @@ message Channel {
     }
 }
 
+message PCost {
+    uint64 cost = 1;
+    int32 iterations = 2;
+}
+
 message ListChannelWithRandom {
     repeated Channel channels = 1;
     bytes randomState = 2 [(scalapb.field).type = "coop.rchain.crypto.hash.Blake2b512Random"];
+    // cost of performing the spatial match
+    PCost cost = 3 [(scalapb.field).no_box = true];
```

Comment:
Who pays for interacting with the contract? If the client then we should charge for `produce` as well

Reply:
No, I mean that we use ListChannelWithRandom to store the data written for a `produce`, and there's no need to supply a PCost in that case.

Reply:
Well, we don't need to persist it but to get the cost for performing a spatial match we need to put it in `ListChannelWithRandom` because that's the API of rspace: https://github.com/rchain/rchain/blob/2c231c3fb7c2350ce2939758c3659ad2f4eb7914/rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/implicits.scala#L32

Reply:
I will change it to boxed type but we didn't have to supply it even when it wasn't boxed -- proto was using `PCost.defaultInstance` (probably used default values for both types of the `PCost`) if not passed explicitly.